LoopKit/GlucoseKit/GlucoseStore.swift

```diff
@@ -111,26 +115,42 @@ public final class GlucoseStore: HealthKitSampleStore {
 
         self.cacheStore = cacheStore
         self.momentumDataInterval = momentumDataInterval
+        
         self.cacheLength = cacheLength
         self.observationInterval = observationInterval ?? cacheLength
 
         super.init(healthStore: healthStore, observeHealthKitSamplesFromOtherApps: observeHealthKitSamplesFromOtherApps, type: glucoseType, observationStart: Date(timeIntervalSinceNow: -self.observationInterval), observationEnabled: observationEnabled)
 
         cacheStore.onReady { (error) in
-            self.dataAccessQueue.async {
-                self.updateLatestGlucose()
+            cacheStore.fetchAnchor(key: GlucoseStore.queryAnchorMetadataKey) { (anchor) in
+                self.dataAccessQueue.async {
+                    self.queryAnchor = anchor
+                    
+                    if !self.authorizationRequired {
+                        self.createQuery()
+                    }
+                    
+                    self.updateLatestGlucose()
+                }
             }
         }
     }
 
     // MARK: - HealthKitSampleStore
+    
+    override func queryAnchorDidChange() {
+        self.log.default("GlucoseStore.queryAnchorDidChange(): %{public}@", String(describing: self.queryAnchor))
```

Comment:
[nit] No similar log statement in either CarbStore or InsulinDeliveryStore. Consider removing here or adding in both other stores.